javascript/ql/src/semmle/javascript/frameworks/NoSQL.qll

```diff
@@ -184,8 +212,221 @@ private module Mongoose {
   /**
    * A Mongoose collection object.
    */
-  class Model extends MongoDB::Collection {
+  class Model extends DataFlow::SourceNode {
```

Comment:
I'm curious why you decided to separate the MongoDB and Mongoose models?

Some thoughts:
- The mongoose model is not extensible (only MongoDB without the extra methods).
- Query calls are no longer members of `DatabaseAccess`, although that concept isn't used much
- Mongoose no longer recognizes instances based on TypeScript types (see comment on `CollectionFromType`).

Could you either reconcile the gap between the models or find a way to share these parts again?

Reply:
> I'm curious why you decided to separate the MongoDB and Mongoose models?

They are not the same thing, Mongoose is a ton of sugar on top of MongoDB. While the inheritance is gone, there is still sharing as seen here:

```
      MongoDB::CollectionMethodSignatures::interpretsArgumentAsQuery(name, n)
```

Both of the inner modules are `private`, so extensibility is not really a concern.

I'll look into avoiding the `DatabaseAccess` and `CollectionFromType` regressions.